gradio/routes.py

```diff
@@ -664,8 +655,14 @@ async def queue_data(
             request: fastapi.Request,
             username: str = Depends(get_current_user),
         ):
-            blocks = app.get_blocks()
-            blocks._queue.attach_data(body)
+            from datetime import datetime
+
+            print("recieved data @", datetime.now().strftime("%H:%M:%S.%f")[:-3])
+            if blocks._queue.server_app is None:
+                blocks._queue.set_server_app(app)
+
+            event_id = await blocks._queue.push(body, request, username)
```

Comment:
wdym by "connect after a job finishes"? Specifically if you try to connect to the SSE stream when there are no pending events, it will throw an error. But the client doesn't let the user arbitrarily connect to the stream endpoint.